docs/src/main/asciidoc/feature-management.adoc

```diff
@@ -0,0 +1,279 @@
+
+==== Feature Flag Check
+
+`FeatureManager` is a `@Bean` that can be `@Autowired` or injected into `@Component` type objects. `FeatureManager` has a method `isEnabledAsync` that, when passed the name of a Feature Flag returns its state.
+
+[source,java,indent=0]
+----
+@Autowired
+FeatureManager featureManager;
+
+if(featureManager.isEnabledAsync("feature-t").block()) {
+    // Do Something
+}
+----
+
+If Feature Management was never configured `isEnabledAsync` or the Feature Flag doesn't exist, it will always return false. If an existing Feature Flags is configured with an unknown Feature Filter, then a `FilterNotFoundException` is thrown. This can be changed to return false by configuring `fail-fast` to false.
+
+[%autowidth.stretch]
+[options="header"]
+|=== 
+|Name | Description | Required | Default
+|spring.cloud.azure.feature.management.fail-fast | Whether throw RuntimeException or not when exception occurs | No |  true
+|===
+
+`FeatureManagerSnapshot` works the same as `FeatureManager`, besides its caching of results in the `@RequestScope`.
+
+==== Feature Gate
+
+With the Feature Management Web library, you can require that a given feature is enabled in order to execute an endpoint. This can be done by using the `@FeatureGate` annotation.
+
+[source,java,indent=0]
+----
+@GetMapping("/featureT")
+@FeatureGate(feature = "feature-t")
+@ResponseBody
+public String featureT() {
+    ...
+}
+----
+
+The `featureT` endpoint can only be accessed if "feature-t" is enabled.
+
+===== Disabled Action Handling
+
+When an endpoint is blocked because the feature it specifies is disabled, `IDisabledFeaturesHandler` will be invoked. By default, an HTTP 404 is returned. This can be overridden by implementing `IDisabledFeaturesHandler`.
+
+[source,java,indent=0]
+----
+@Component
+public class DisabledFeaturesHandler implements IDisabledFeaturesHandler{
+
+    @Override
+    public HttpServletResponse handleDisabledFeatures(HttpServletRequest request, HttpServletResponse response) {
+        ...
+        return response;
+    }
+
+}
+----
+
+===== Routing
+
+Certain routes may expose application capabilities that are gated by features. These routes can be redirected if a feature is disabled to another endpoint.
+
+[source,java,indent=0]
+----
+@GetMapping("/featureT")
+@FeatureGate(feature = "feature-t" fallback= "/oldEndpoint")
+@ResponseBody
+public String featureT() {
+    ...
+}
+
+@GetMapping("/oldEndpoint")
+@ResponseBody
+public String oldEndpoint() {
+    ...
+}
+----
+
+=== Built-In Feature Filters
+
+There are a few feature filters that come with the `azure-spring-cloud-feature-management` package. These feature filters are not added automatically, but can be setup in an `@Configuration` for use.
+
+==== AlwaysOnFilter
+
+This filter always returns true.
+
+==== PercentageFilter
+
+This filter provides the capability to enable a feature based on a set percentage.
```

Comment:
One question we get a lot is who this applies to - so I would add a few words that it enables the feature for everyone that has access to the filter, and that it is not sticky aka it randomly applies the percentage to a request, it does not remember who you are as a user and is not guaranteed to give you the same on or off experience consistently